### PR TITLE
SIG-Scheduling 2020 Annual Report

### LINKED ISSUE #5510: 2021 Annual Report: SIG Scheduling
Chairs: @Huang-Wei @ahg-g 
Liasion: @liggitt

- [ ] Consult SIG to complete draft of report
- [ ] If needed, consult with Steering Committee liasion on any private issues or concerns that arrise while completing report
- [ ] Submit PR to git.k8s.io/community repo with copy of report, assigning your SC liasion for review
- [ ] Steering Committee reviews report, and may make comments/ask questions
- [ ] Once all comments are addressed, SC will approve report to merge
- [ ] If needed, any follow up items may be brought to the "Chairs, Tech Leads, and Organizers" meeting in April

Once all the above items are complete, this issue may be `/close`'d

Key dates:
- Initial PR to communtiy repo should be opened by March 1st, 2021
- PR should be reviewed and merged by April 1st, 2021

More detailed information on the annual reports process is available [here](https://git.k8s.io/community/committee-steering/governance/annual-reports.md).

If you have any questions or concerns about this process, you may reach the Steering Committee via the following methods:
- Slack/E-mail your liasion directly
- The public #steering-committee channel in Slack
- The public mailing list steering@kubernetes.io
- The private mailing list steering-private@kubernetes.io for any private or sensitive issues.

/sig scheduling
/committee steering

### PR DESCRIPTION
**Pull request by Huang-Wei**
Adding initial template for SIG-Scheduling 2020 Annual Report.

This is due by March 8th and will be shown to the SIG and Slack.

**Which issue(s) this PR fixes**:

Fixes #5510.

### CODE REVIEW

**Comment by Huang-Wei** [comment]
Opened this as a placeholder, @ahg-g and I will fill out unanswered parts by this week.

/assign @liggitt

File: sig-scheduling/2020-annual-report.md
Comment (ahg-g) [inline sig-scheduling/2020-annual-report.md:73]: - Alpha:
  -  [Prefer Nominated Node](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/1923-prefer-nominated-node)
  - [Node Resource Strategy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/2458-node-resource-score-strategy)
  - [Pod Affinity Namespace Selector](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/2249-pod-affinity-namespace-selector)
- Beta:
  - [Default topology spread](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/1258-default-pod-topology-spread) (will graduate with CC)
  - [Multi Scheduling Profiles](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/1451-multi-scheduling-profiles)
  - [Non-preempting priority class](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass)
  - [Component Config](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/785-scheduler-component-config-api)

**Review by ahg-g — COMMENTED** [review]
some suggestions

**Comment by Huang-Wei** [comment]
@liggitt the first version is ready for review. Take your time.

File: sig-scheduling/2020-annual-report.md
Comment (cofyc) [inline sig-scheduling/2020-annual-report.md:90]: - [Volume Capacity Priority] co-owned by sig-storage

Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:90]: done.

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:25]: is this something contribex can help with?

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:38]: which ones are inactive?

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:48]: are you growing more reviewers and approvers?

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:56]: Do you use help-wanted or good-first issue labels?

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:66]: Do you want more companies to contribute? If so, to what and what extent? This is a microphone to end users

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:73]: ty for sectioning them out this way

File: sig-scheduling/2020-annual-report.md
Comment (parispittman) [inline sig-scheduling/2020-annual-report.md:100]: ty for calling this out

**Review by parispittman — CHANGES_REQUESTED** [review]

File: sig-scheduling/2020-annual-report.md
Comment (ahg-g) [reply sig-scheduling/2020-annual-report.md:25]: I think we are fine, we can send a PR to remove inactive members and tag them on that PR.

File: sig-scheduling/2020-annual-report.md
Comment (ahg-g) [reply sig-scheduling/2020-annual-report.md:48]: I think we have have enough approvals for the volume of the PRs we are getting, reviewers are growing.

File: sig-scheduling/2020-annual-report.md
Comment (ahg-g) [reply sig-scheduling/2020-annual-report.md:56]: both depending on the complexity of the issue.

File: sig-scheduling/2020-annual-report.md
Comment (ahg-g) [reply sig-scheduling/2020-annual-report.md:66]: I don't think we are lacking diversity, there are 5+ companies involved in the sig on/off.

File: sig-scheduling/2020-annual-report.md
Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:38]: - [kube-batch]. The original plan is to develop batch-related features in a separate project, and eventually converge with default scheduler. But kube-batch is now incubated into a cncf sandobx project called volcano. So kube-batch is not active now.
- [poseidon]. I don't see much interested in this project, and it's also no longer actively developed.

[kube-batch]: https://github.com/kubernetes-sigs/kube-batch
[poseidon]: https://github.com/kubernetes-sigs/poseidon

File: sig-scheduling/2020-annual-report.md
Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:25]: I suppose it's a common issue across all sigs. It'd be neat if we can come up with a script auto-identifying the inactive maintainers/reviewers:

- for a particular SIG, not participated in any PR reviews in the last 12 (or 18) months
- a single bare "/lgtm" or "/approve" doesn't count as a legal review

File: sig-scheduling/2020-annual-report.md
Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:48]: Yes, we're fine by now.

File: sig-scheduling/2020-annual-report.md
Comment (liggitt) [reply sig-scheduling/2020-annual-report.md:38]: are those subprojects inactive to the point that their readme's should be updated to note that?

File: sig-scheduling/2020-annual-report.md
Comment (liggitt) [inline sig-scheduling/2020-annual-report.md:109]:
```suggestion
We haven't started leveraging devstat data or Github board to get a high-level picture of PR/Issue.
```

File: sig-scheduling/2020-annual-report.md
Comment (liggitt) [reply sig-scheduling/2020-annual-report.md:100]: agreed, thanks. I noticed https://docs.google.com/document/d/1t2QoASmg_BssFczl8n7UNTDWKgncMSrLTX802fQ0lK8/edit# and https://github.com/kubernetes/community/pull/5674 recently, which seemed to address some of the developer-facing bits. Would it make sense to open help-wanted issues for remaining dev doc gaps and for specific user-oriented gaps?

**Review by liggitt — COMMENTED** [review]
lgtm, I just had one question about the two inactive subprojects mentioned

File: sig-scheduling/2020-annual-report.md
Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:100]: Yes, @ingvagabund 's  recent contributions really make the docs greatly improved. With those docs in place, I think we're fine on dev docs. In terms of user-oriented doc gaps, we will open help-wanted issues for tracking.

File: sig-scheduling/2020-annual-report.md
Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:38]: I will create issues in those subprojects to discuss what to do next: whether specify the inactive status in README, or just archive them.

Comment (Huang-Wei) [reply sig-scheduling/2020-annual-report.md:38]: Followup for:

- kube-batch: https://github.com/kubernetes-sigs/kube-batch/issues/948
- poseidon: https://github.com/kubernetes-sigs/poseidon/issues/187

**Comment by liggitt** [comment]
/lgtm
with steering hat

/hold
for lgtm by sig-scheduling co-lead @ahg-g

File: sig-scheduling/2020-annual-report.md
Comment (lilinji) [reply sig-scheduling/2020-annual-report.md:25]: ok

**Pull request merged**
